src/Infrastructure/Persistence/Contexts/BaseDbContext.cs

```diff
@@ -39,30 +40,28 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
         modelBuilder.ApplyGlobalFilters<ISoftDelete>(s => s.DeletedOn == null);
     }
 
-    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
-    {
-        optionsBuilder.EnableSensitiveDataLogging();
-
-        string tenantConnectionString = _tenantService.GetConnectionString();
-        if (!string.IsNullOrEmpty(tenantConnectionString))
-        {
-            string dbProvider = _tenantService.GetDatabaseProvider();
-            switch (dbProvider.ToLower())
-            {
-                case "postgresql":
-                    optionsBuilder.UseNpgsql(tenantConnectionString);
-                    break;
-
-                case "mssql":
-                    optionsBuilder.UseSqlServer(tenantConnectionString);
-                    break;
-
-                case "mysql":
-                    optionsBuilder.UseMySql(tenantConnectionString, ServerVersion.AutoDetect(tenantConnectionString));
-                    break;
-            }
-        }
-    }
+    // is this still necessary with how multitenancy is now implemented?
```

Comment:
This part should be tested since the same thing I was thinking, also the question arises, can a client have a provider other than root ???

Reply:
I think this is still needed for migrations to work. Have to test this.

Reply:
No the way it's implemented now apparently, all databases have to have the same provider. I.e. the provider is only relevant on the root level for now.
I'll probably leave this code in for this PR. Another PR can look into removing and testing this ;-)